app/models/miq_vim_broker_worker.rb

```diff
@@ -1,13 +1,17 @@
 class MiqVimBrokerWorker < MiqWorker
   require_nested :Runner
 
-  self.required_roles         = %w(
-    ems_inventory
-    ems_metrics_collector
-    ems_operations
-    smartproxy
-    smartstate
-  )
+  self.required_roles         = lambda {
+    %w(
+      ems_metrics_collector
+      ems_operations
+      smartproxy
+      smartstate
+    ).tap do |roles|
+      roles << 'ems_inventory' unless Settings.prototype.try(:ems_vmware).try(:update_driven_refresh)
```

Comment:
The point of this is to control if the broker should start or not, the check in the runner is to control the cache_scope once it has started.  Some examples would probably help.

Assuming you're using `update_driven_refresh`:
1. The only active role on this appliance is `ems_inventory`
In this case the broker shouldn't even start, that's what the above block handles

2. There are e.g. ems_operations and ems_inventory roles active
In this case we do want to start the broker to handle the ems_operations role but without checking the setting in the runner it would use `cache_scope_ems_refresh` which uses a lot more memory and defeat the purpose.  This is what the check in the runner handles.